src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NullablePublicOnly.cs

```diff
@@ -195,5 +195,154 @@ private static void AssertAttributes(ImmutableArray<CSharpAttributeData> attribu
             var actualNames = attributes.Select(a => a.AttributeClass.ToTestDisplayString()).ToArray();
             AssertEx.SetEqual(actualNames, expectedNames);
         }
+
+        [Fact]
+        [WorkItem(36457, "https://github.com/dotnet/roslyn/issues/36457")]
+        public void ExplicitAttribute_ReferencedInSource_Assembly()
+        {
+            var sourceAttribute =
+@"namespace System.Runtime.CompilerServices
+{
+    internal class NullablePublicOnlyAttribute : System.Attribute
+    {
+        internal NullablePublicOnlyAttribute(bool b) { }
+    }
+}";
+            var source =
+@"using System.Runtime.CompilerServices;
+[assembly: NullablePublicOnly(false)]
```

Comment:
Should we get an error on this attribute?

Reply:
Only if `AttributeTargets` is set.

---
In reply to: [296932588](https://github.com/dotnet/roslyn/pull/36152#discussion_r296932588) [](ancestors = 296932588)